cli/src/commands/task/take_more_labels.rs

```diff
@@ -0,0 +1,258 @@
+    make_progress()?;
+
+    for source_id in target_dataset.source_ids.iter() {
+        for target_comment_batch in
+            client.get_labellings_iter(&target_dataset.full_name(), source_id, false, None)
+        {
+            let target_comment_batch = target_comment_batch.chain_err(|| {
+                ErrorKind::Client("Getting batch of target comments failed.".into())
+            })?;
+            statistics.add_target_comments(target_comment_batch.len());
+
+            let mut target_labellings_to_update =
+                get_labellings_with_assigned_label(target_comment_batch, &only_for_target_label);
+
+            if target_labellings_to_update.is_empty() {
+                continue;
+            }
+
+            // Fetch source labellings for our given ids
+            let source_annotated_comments = client
+                .get_labellings(
+                    &source_dataset.full_name(),
+                    target_labellings_to_update.keys(),
+                )
+                .chain_err(|| {
+                    ErrorKind::Client("Getting batch of source labellings failed.".into())
+                })?;
+
+            for source_annotated_comment in source_annotated_comments.into_iter() {
+                let AnnotatedComment {
+                    comment: Comment { uid, .. },
+                    labelling: source_labelling,
+                    ..
+                } = source_annotated_comment;
+                let source_assigned = source_labelling.unwrap_or_default().assigned;
+                if source_assigned.is_empty() {
+                    continue;
+                }
+                let Labelling {
+                    assigned: target_assigned,
+                    dismissed,
+                    ..
+                } = target_labellings_to_update
+                    .remove(&uid)
+                    .expect("Tried to make a new labelling for a comment we don't need to update.");
+
+                // Merge assigned labels from the two datasets
+                let assigned = merge_assigned_labels(
+                    target_assigned,
+                    source_assigned,
+                    &transferred_label_parent,
+                );
+                let new_labelling = NewLabelling {
+                    assigned,
+                    dismissed,
+                };
+
+                client
+                    .update_labelling(
+                        &target_dataset.full_name(),
+                        &uid,
+                        Some(&new_labelling),
+                        None,
+                    )
+                    .chain_err(|| ErrorKind::Client("Updating labelling failed.".into()))?;
+                statistics.add_labellings_updated(1);
+            }
+
+            let _progress = make_progress()?;
+        }
+    }
+    Ok(())
+}
+
+/// Collect target comment ids and labellings, which contain the given assigend label.
+fn get_labellings_with_assigned_label(
```

Comment:
This function name is a bit weird, isn't it really `map_comments_to_assigned_labels`?